src/js/select2/selection/search.js

```diff
@@ -175,7 +175,7 @@ define([
 
     this.resizeSearch();
     if (searchHadFocus) {
-      this.$search.focus();
+      this.$element.focus();
```

Comment:
Will this have side effects on other types of inputs? I did another PR which only modifies the behaviour for tag input as a precaution. https://github.com/select2/select2/pull/4958/files

Reply:
It seems that there are several competing solutions to this problem:

https://github.com/select2/select2/issues/4398#issuecomment-281721604
https://github.com/select2/select2/pull/4958/files
https://github.com/select2/select2/commit/294e7e664d95eafb7f41264c2b87412ad9aa48fc